README.md

```diff
@@ -1,15 +1,91 @@
-# Unified Distributed Execution
+<p align="center"><a href="https://unidist.readthedocs.io"><img width=77% alt="" src="https://github.com/modin-project/unidist/blob/master/docs/img/unidist_logo.png?raw=true"></a></p>
+<h2 align="center">Unified Distributed Execution</h2>
 
-The framework supports multiple execution backends: Ray, Dask, MPI and MultiProcessing.
+<p align="center">
+<a href="https://github.com/modin-project/unidist/actions"><img src="https://github.com/modin-project/unidist/workflows/master/badge.svg" align="center"></a>
+<a href="https://unidist.readthedocs.io/en/latest/?badge=latest"><img alt="" src="https://readthedocs.org/projects/unidist/badge/?version=latest" align="center"></a>
+</p>
 
-To run tests you need to install ``pytest`` package:
+### What is unidist?
 
+unidist is a framework that is intended to provide the unified API for distributed execution by supporting various performant execution backends and utilizing all of your CPUs cores. At the moment the following backends
+are supported underneath the hood:
+
+* [Ray](https://docs.ray.io/en/master/index.html)
+* [MPI](https://www.mpi-forum.org/)
+* [Dask Distributed](https://distributed.dask.org/en/latest/)
+* [Python Multiprocessing](https://docs.python.org/3/library/multiprocessing.html)
+
+unidist is designed to work in a [task-based parallel](https://en.wikipedia.org/wiki/Task_parallelism) model.
+
+Also, the framework provides a sequential ``Python`` backend, that can be used for debugging.
+
+### Instalation
+
+Currently, unidist can be installed from sources using ``pip``:
+
+```bash
+$ pip install git+https://github.com/modin-project/unidist
 ```
-pip install pytest
+
+This will install unidist directly from the repo without you having to manually clone it! Please be aware that these changes have not made it into a release and may not be completely stable.
+
+#### Choosing an execution backend
+
+There are several ways to choose the execution backend for distributing computations.
+First, the recommended way is to use
+[unidist CLI](https://unidist.readthedocs.io/en/latest/using_cli.html) options:
+
+```bash
+# Running the script with unidist on Ray backend
+$ unidist script.py --backend ray
+# Running the script with unidist on MPI backend
+$ unidist script.py --backend mpi
 ```
-Run tests using command:
+
+Second, setting the environment variable:
+
+```bash
+# unidist will use Ray backend to distribute computations
+export UNIDIST_BACKEND=ray
+# unidist will use MPI backend to distribute computations
+export UNIDIST_BACKEND=mpi
 ```
-pytest unidist/test/
+
+Third, using [config API](https://unidist.readthedocs.io/en/latest/flow/unidist/config.html) directly in your script:
+
+```python
+import unidist.config as cfg
+cfg.Backend.put("mpi") # unidist will use MPI backend to distribute computations
```

Comment:
Need to add note, that MPI backend must be run via `unidist script.py -b mpi` always with why we do that.

Reply:
We should put this note in using_mpi section so for now changed it to Dask usage to simplify Readme. Once we have more stable MPI and more docs on it, we will be able to expose MPI in Readme.

Reply:
ok, will see in https://github.com/modin-project/unidist/issues/85